src/wrappers/themis/react-native-themis/android/src/main/java/com/reactnativethemis/ThemisModule.java

```diff
@@ -0,0 +1,640 @@
+
+  private final static int comparator_not_ready = 0;
+  private final static int comparator_not_match = 22;
+  private final static int comparator_match = 21;
+  private final static int comparator_error = -1;
+
+  private final static String contextRequired = "Context required";
+  private final static String privateKeyRequired = "Private key can not be null or blank";
+  private final static String publicKeyRequired = "Public key can not be null or blank";
+
+  ThemisModule(ReactApplicationContext context)
+  {
+    super(context);
+  }
+
+  @NonNull
+  @Override
+  public String getName()
+  {
+    return "Themis";
+  }
+
+  /* Replace JAVA enum version to equal with iOS version. */
+
+  @Override
+  public Map<String, Object> getConstants()
+  {
+    final Map<String, Object> constants = new HashMap<>();
+    constants.put("COMPARATOR_NOT_READY", comparator_not_ready);
+    constants.put("COMPARATOR_NOT_MATCH", comparator_not_match);
+    constants.put("COMPARATOR_MATCH", comparator_match);
+    constants.put("COMPARATOR_ERROR", comparator_error);
+    constants.put("KEYTYPE_RSA", AsymmetricKey.KEYTYPE_RSA);
+    constants.put("KEYTYPE_EC", AsymmetricKey.KEYTYPE_EC);
+    return constants;
+  }
+
+  private static int compareResultSerialize(SecureCompare.CompareResult result)
+  {
+    switch (result) {
+      case NOT_READY: return comparator_not_ready;
+      case NO_MATCH: return comparator_not_match;
+      case MATCH: return comparator_match;
+      default: return comparator_error;
+    }
+  }
+  private static WritableArray dataSerialize(byte[] data)
+  {
+    if (data == null) {
+      return null;
+    }
+    WritableArray result = new WritableNativeArray();
+    for (byte datum : data) result.pushInt(datum);
+
+    return result;
+  }
+
+  private static byte[] dataDeserialize(ReadableArray serializedData) throws ByteOverflowException {
+    if (serializedData == null || serializedData.size() == 0) {
+      return null;
+    }
+    byte[] data = new byte[serializedData.size()];
+    for (int i = 0; i < serializedData.size(); i++) {
+      if(serializedData.getInt(i) >= -128 && serializedData.getInt(i) <= 255) {
+        byte j = (byte) serializedData.getInt(i);
+        data[i] = j;
+      } else {
+        throw new ByteOverflowException("Value " + serializedData.getInt(i) + " is out of range");
+      }
+    }
+    return data;
+  }
+
+  @ReactMethod
+  public void stringSerialize(String text,
+                              Callback callback)
+  {
+    byte[] data = text.getBytes(StandardCharsets.UTF_8);
+    WritableArray response = dataSerialize(data);
+    callback.invoke(response);
+  }
+
+  @ReactMethod
+  public void keyPair(int algorithm, Callback callback)
+  {
+    Keypair pair;
+    switch (algorithm) {
+      case AsymmetricKey.KEYTYPE_RSA:
+        pair = KeypairGenerator.generateKeypair(AsymmetricKey.KEYTYPE_RSA);
+        break;
+      default:
+        pair = KeypairGenerator.generateKeypair();
```

Comment:
```suggestion
        pair = KeypairGenerator.generateKeypair(AsymmetricKey.KEYTYPE_EC);
```

should we put default value here?

Reply:
We can do it. Sure. For expllicity.